// starknet-types-07/dist/cjs/api/nonspec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ETransactionVersion3 = exports.ETransactionVersion2 = exports.ETransactionVersion = exports.EDAMode = exports.EDataAvailabilityMode = exports.EBlockTag = exports.ETransactionExecutionStatus = exports.ETransactionFinalityStatus = exports.ETransactionStatus = exports.ESimulationFlag = exports.ETransactionType = void 0;
var ETransactionType;
(function (ETransactionType) {
    ETransactionType["DECLARE"] = "DECLARE";
    ETransactionType["DEPLOY"] = "DEPLOY";
    ETransactionType["DEPLOY_ACCOUNT"] = "DEPLOY_ACCOUNT";
    ETransactionType["INVOKE"] = "INVOKE";
    ETransactionType["L1_HANDLER"] = "L1_HANDLER";
})(ETransactionType || (exports.ETransactionType = ETransactionType = {}));
var ESimulationFlag;
(function (ESimulationFlag) {
    ESimulationFlag["SKIP_VALIDATE"] = "SKIP_VALIDATE";
    ESimulationFlag["SKIP_FEE_CHARGE"] = "SKIP_FEE_CHARGE";
})(ESimulationFlag || (exports.ESimulationFlag = ESimulationFlag = {}));
var ETransactionStatus;
(function (ETransactionStatus) {
    ETransactionStatus["RECEIVED"] = "RECEIVED";
    ETransactionStatus["REJECTED"] = "REJECTED";
    ETransactionStatus["ACCEPTED_ON_L2"] = "ACCEPTED_ON_L2";
    ETransactionStatus["ACCEPTED_ON_L1"] = "ACCEPTED_ON_L1";
})(ETransactionStatus || (exports.ETransactionStatus = ETransactionStatus = {}));
var ETransactionFinalityStatus;
(function (ETransactionFinalityStatus) {
    ETransactionFinalityStatus["ACCEPTED_ON_L2"] = "ACCEPTED_ON_L2";
    ETransactionFinalityStatus["ACCEPTED_ON_L1"] = "ACCEPTED_ON_L1";
})(ETransactionFinalityStatus || (exports.ETransactionFinalityStatus = ETransactionFinalityStatus = {}));
var ETransactionExecutionStatus;
(function (ETransactionExecutionStatus) {
    ETransactionExecutionStatus["SUCCEEDED"] = "SUCCEEDED";
    ETransactionExecutionStatus["REVERTED"] = "REVERTED";
})(ETransactionExecutionStatus || (exports.ETransactionExecutionStatus = ETransactionExecutionStatus = {}));
var EBlockTag;
(function (EBlockTag) {
    EBlockTag["LATEST"] = "latest";
    EBlockTag["PENDING"] = "pending";
})(EBlockTag || (exports.EBlockTag = EBlockTag = {}));
var EDataAvailabilityMode;
(function (EDataAvailabilityMode) {
    EDataAvailabilityMode["L1"] = "L1";
    EDataAvailabilityMode["L2"] = "L2";
})(EDataAvailabilityMode || (exports.EDataAvailabilityMode = EDataAvailabilityMode = {}));
var EDAMode;
(function (EDAMode) {
    EDAMode[EDAMode["L1"] = 0] = "L1";
    EDAMode[EDAMode["L2"] = 1] = "L2";
})(EDAMode || (exports.EDAMode = EDAMode = {}));
var ETransactionVersion;
(function (ETransactionVersion) {
    ETransactionVersion["V0"] = "0x0";
    ETransactionVersion["V1"] = "0x1";
    ETransactionVersion["V2"] = "0x2";
    ETransactionVersion["V3"] = "0x3";
    ETransactionVersion["F0"] = "0x100000000000000000000000000000000";
    ETransactionVersion["F1"] = "0x100000000000000000000000000000001";
    ETransactionVersion["F2"] = "0x100000000000000000000000000000002";
    ETransactionVersion["F3"] = "0x100000000000000000000000000000003";
})(ETransactionVersion || (exports.ETransactionVersion = ETransactionVersion = {}));
var ETransactionVersion2;
(function (ETransactionVersion2) {
    ETransactionVersion2["V0"] = "0x0";
    ETransactionVersion2["V1"] = "0x1";
    ETransactionVersion2["V2"] = "0x2";
    ETransactionVersion2["F0"] = "0x100000000000000000000000000000000";
    ETransactionVersion2["F1"] = "0x100000000000000000000000000000001";
    ETransactionVersion2["F2"] = "0x100000000000000000000000000000002";
})(ETransactionVersion2 || (exports.ETransactionVersion2 = ETransactionVersion2 = {}));
var ETransactionVersion3;
(function (ETransactionVersion3) {
    ETransactionVersion3["V3"] = "0x3";
    ETransactionVersion3["F3"] = "0x100000000000000000000000000000003";
})(ETransactionVersion3 || (exports.ETransactionVersion3 = ETransactionVersion3 = {}));
